extension/src/popup/views/SignAuthEntry/index.tsx

```diff
@@ -77,6 +83,23 @@ export const SignAuthEntry = () => {
     );
   }
 
+  if (entryNetworkPassphrase && entryNetworkPassphrase !== networkPassphrase) {
+    return (
+      <WarningMessage
+        variant={WarningMessageVariant.warning}
+        handleCloseClick={() => window.close()}
+        isActive
+        header={`${t("Freighter is set to")} ${networkName}`}
+      >
+        <p>
+          {t("The requester expects you to sign this auth entry on")}{" "}
+          {entryNetworkPassphrase}.
+        </p>
+        <p>{t("Signing this transaction is not possible at the moment.")}</p>
+      </WarningMessage>
+    );
+  }
+
```

Comment:
now that signAuthEntry and signMessage accept a desired network, they should also show a warning when the user is on the wrong network. Lifted from the SignTransaction UI